diff --git a/3rdparty/util-linux-libs-orig/config/config.sub b/3rdparty/util-linux-libs-workdir/config/config.sub
index b51fb8c..40acd25 100755
--- a/3rdparty/util-linux-libs-orig/config/config.sub
+++ b/3rdparty/util-linux-libs-workdir/config/config.sub
@@ -1360,7 +1360,7 @@ case $os in
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd*)
+	     | midnightbsd* | vinix*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
